glue_runner.py

```diff
@@ -0,0 +1,276 @@
+    1. Print Usage Instructions and Exit:
+        $ python glue_runner.py -h
+    2. Create a new Glue job and run your first script in us-east-1 (the default region):
+        $ python glue_runner.py deltacat/examples/hello_world.py \
+          --glue-iam-role "AWSGlueServiceRole" \
+    3. Run an example in us-east-1 using the last job config and DeltaCAT deploy:
+        $ python glue_runner.py deltacat/examples/hello_world.py
+    4. Run an example in us-east-1 using your local workspace copy of DeltaCAT:
+        $ python glue_runner.py deltacat/examples/hello_world.py --deploy-local-deltacat
+       Note that the deployed package is referenced by an S3 URL that expires in 7 days.
+       After 7 days, you must deploy a new DeltaCAT package to avoid receiving a 403 error!
+    5. Create a new job and run an example in us-west-2:
+        $ python glue_runner.py deltacat/examples/hello_world.py \
+          --region us-west-2
+          --glue-iam-role "AWSGlueServiceRole" \
+    6. Pass arguments into an example script as environment variables:
+        $ python glue_runner.py deltacat/examples/logging.py\
+          --example-script-args '{"--var1":"Try that", "--var2":"DeltaCAT"}' \
+
+During execution this script will:
+    1. Create a S3 bucket at `s3://deltacat-packages-$USER` if it doesn't already exist.
+    2. [Optional] Build and deploy your local DeltaCAT workspace to s3://deltacat-packages-$USER/.
+    3. Create a S3 bucket at `s3://deltacat-glue-scripts-$USER` if it doesn't already exist.
+    4. Upload deltacat/examples/$EXAMPLE_SCRIPT_NAME to s3://deltacat-glue-scripts-$USER.
+    5. Create or update the Glue Job "deltacat-example-runner-$USER" to run this example.
+    6. Run the "deltacat-example-runner-$USER" Glue Job with either the newly built DeltaCAT wheel or the last used wheel.
+
+Where every instance of $USER will be replaced by the value of `os.environ["USER"]`.
+
+"""
+import os
+import argparse
+
+parser = argparse.ArgumentParser()
+parser.add_argument(
+    "script_path",
+    help="Path to the script to run (e.g. 'deltacat/examples/hello_world.py').",
+    type=str,
+)
+parser.add_argument(
+    "-a",
+    "--example-script-args",
+    help="JSON dictionary of arguments to set when running the example script.",
+    type=str,
+)
+parser.add_argument(
+    "-r",
+    "--region",
+    help="AWS region to setup (default: us-east-1).",
+    default="us-east-1",
+    type=str,
+)
+parser.add_argument(
+    "-i",
+    "--glue-iam-role",
+    help="Name of IAM role onboarded for use with Glue.",
+    type=str,
+)
+parser.add_argument(
+    "-w",
+    "--glue-max-workers",
+    help="Maximum number of glue workers (default: 25).",
+    default=25,
+    type=int,
+)
+parser.add_argument(
+    "-n",
+    "--glue-job-name",
+    help="Name of the Glue Job to create/update (default: deltacat-example-runner-$USER).",
+    type=str,
+)
+args_mutex = parser.add_mutually_exclusive_group()
+args_mutex.add_argument(
```

Comment:
Does this script also handle deploying the custom implementation of deltacat storage interface? Because, without that all APIs yield `NotImplementedError` correct?

Reply:
This script will handle deploying implementations as they're made available in subsequent PRs, but that's correct - all storage APIs yield `NotImplementedError` with this PR merged today.